### PR TITLE
[Validator] Add docs for "canonicalize" option at `Locale` validation constraint

### LINKED ISSUE #7660: [RFC] [Validator] Suggest to apply `\Locale::canonicalize()` before validating with `Locale` constraint
It would be nice to have a note about [`\Locale::canonicalize()`](http://php.net/manual/en/locale.canonicalize.php) within the `Locale` constraint documentation, since many times locales are expressed using other "uncanonicalized" formats. By instance, a common use case is to infer the locale from javascript, where `navigator.language` returns values like "en-US" instead of "en_US"; making the `Locale` constraint to fail.
The canonicalization process is explained at [ICU project](http://userguide.icu-project.org/locale#TOC-Canonicalization).

### PR DESCRIPTION
**Pull request by phansys**
Related to symfony/symfony#22353, symfony/symfony#26075.

Closes #7660.

### CODE REVIEW

File: reference/constraints/Locale.rst [hunk 1]
```diff
@@ -90,5 +96,18 @@ This message is shown if the string is not a valid locale.
 
 .. include:: /reference/constraints/_payload-option.rst.inc
 
+canonicalize
+~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``false``
+
+.. versionadded:: 4.1
+    Using this option with value `false` was deprecated in Symfony 4.1 and it will
```
Comment (javiereguiluz) [inline reference/constraints/Locale.rst]: In this `versionadded` directive, the `false` and `false` code have single backticks but the RST format requires double backticks

File: reference/constraints/Locale.rst [hunk 2]
```diff
@@ -90,5 +96,18 @@ This message is shown if the string is not a valid locale.
...
+    throw an exception in Symfony 5.0, use `true` instead.
+
+If true, then the :phpmethod:`Locale::canonicalize` method will be applied to
```
Comment (javiereguiluz) [inline reference/constraints/Locale.rst]: `If true, then the ...` -> `If true, the ...` ?

Comment (javiereguiluz) [inline reference/constraints/Locale.rst]: `applied to before...` -> `applied before...`

File: reference/constraints/Locale.rst [hunk 3]
```diff
@@ -90,5 +96,18 @@ This message is shown if the string is not a valid locale.
...
+before checking the validity of the given locale.
```
Comment (javiereguiluz) [inline reference/constraints/Locale.rst]: Let's add a simple example at the end of this phrase for those who don't fully understand what "canonicalize" means or does. Example:

```rst
... before checking the validity of the given locale (e.g. ``FR-fr.utf8``
is transformed into ``fr_FR``).
```

**Review by javiereguiluz — COMMENTED** [review]
Nice! Thanks for this contribution @phansys (and thanks for implementing the feature too). I left some minor comments.

**Comment by phansys** [comment]
Your comments were addressed, thank you @javiereguiluz!

**Comment by javiereguiluz** [comment]
Thanks Javier.

**Pull request merged**